Build failure: corretto11

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

I have a nixos system (x86_64) configured using flakes. I have `corretto11` installed as a user package (via home-manager's `home.packages = [];`).

I've just run this command to update and rebuild my system:

```sh
nix flake update
nixos-rebuild build --flake ~/nix
```

Corretto is consistently running into a build failure that is stopping my system rebuild. The attached log output was gathered via `nixos-rebuild build --flake ~/nix --print-build-logs`.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/320460564

### Relevant log output

```console
corretto11> === Output from failing command(s) repeated here ===
corretto11> * For target buildtools_depend__the.COMPILE_DEPEND_batch:
corretto11> * For target buildtools_interim_rmic_modules_jdk.rmic.interim__the.BUILD_jdk.rmic.interim_batch:
corretto11> * For target buildtools_jdk.vm.compiler.match.processor__the.BUILD_VM_COMPILER_MATCH_PROCESSOR_batch:
corretto11> * For target buildtools_jdk.vm.compiler.nodeinfo.processor__the.BUILD_VM_COMPILER_NODEINFO_PROCESSOR_batch:
corretto11> * For target buildtools_jdk.vm.compiler.options.processor__the.BUILD_VM_COMPILER_OPTIONS_PROCESSOR_batch:
corretto11> * For target buildtools_jdk.vm.compiler.replacements.verifier__the.BUILD_VM_COMPILER_REPLACEMENTS_PROCESSOR_batch:
corretto11> * For target buildtools_jdk.vm.compiler.serviceprovider.processor__the.BUILD_VM_COMPILER_SERVICEPROVIDER_PROCESSOR_batch:
corretto11> * For target buildtools_jdk_tools_classes__the.BUILD_TOOLS_JDK_batch:
corretto11> * For target hotspot_variant-server_libjvm_gtest_objs_precompiled_precompiled.hpp.gch:
corretto11> cc1plus: warning: command-line option '-std=gnu17' is valid for C/ObjC but not for C++
corretto11> In file included from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/utilities/align.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/runtime/globals.hpp:29,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/memory/allocation.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/oops/array.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/memory/universe.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/code/oopRecorder.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/asm/codeBuffer.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/asm/assembler.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/precompiled/precompiled.hpp:31:
corretto11> /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/utilities/globalDefinitions.hpp:1188:28: error: 'unsigned int uabs(int)' was declared 'extern' and later 'static' [-fpermissive]
corretto11>  1188 | static inline unsigned int uabs(int n) { return uabs((unsigned int)n); }
corretto11>       |                            ^~~~
corretto11> In file included from /nix/store/mjf8jlq9grydcdvyw6hb063x5c34g5gf-gcc-15.2.0/include/c++/15.2.0/cstdlib:83,
corretto11>                  from /nix/store/mjf8jlq9grydcdvyw6hb063x5c34g5gf-gcc-15.2.0/include/c++/15.2.0/stdlib.h:36,
corretto11>    ... (rest of output omitted)
corretto11> * For target hotspot_variant-server_libjvm_objs_precompiled_precompiled.hpp.gch:
corretto11> cc1plus: warning: command-line option '-std=gnu17' is valid for C/ObjC but not for C++
corretto11> In file included from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/utilities/align.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/runtime/globals.hpp:29,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/memory/allocation.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/oops/array.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/memory/universe.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/code/oopRecorder.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/asm/codeBuffer.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/asm/assembler.hpp:28,
corretto11>                  from /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/precompiled/precompiled.hpp:31:
corretto11> /build/source/installers/linux/universal/tar/corretto-build/buildRoot/src/hotspot/share/utilities/globalDefinitions.hpp:1188:28: error: 'unsigned int uabs(int)' was declared 'extern' and later 'static' [-fpermissive]
corretto11>  1188 | static inline unsigned int uabs(int n) { return uabs((unsigned int)n); }
corretto11>       |                            ^~~~
corretto11> In file included from /nix/store/mjf8jlq9grydcdvyw6hb063x5c34g5gf-gcc-15.2.0/include/c++/15.2.0/cstdlib:83,
corretto11>                  from /nix/store/mjf8jlq9grydcdvyw6hb063x5c34g5gf-gcc-15.2.0/include/c++/15.2.0/stdlib.h:36,
corretto11>    ... (rest of output omitted)
corretto11>
corretto11> * All command lines available in /build/source/installers/linux/universal/tar/corretto-build/buildRoot/build/linux-x86_64-normal-server-release/make-support/failure-logs.
corretto11> === End of repeated output ===
corretto11>
corretto11> No indication of failed target found.
corretto11> Hint: Try searching the build log for '] Error'.
corretto11> Hint: See doc/building.html#troubleshooting for assistance.
corretto11>
corretto11> make[1]: *** [/build/source/installers/linux/universal/tar/corretto-build/buildRoot/make/Init.gmk:314: main] Error 2
corretto11> make: *** [/build/source/installers/linux/universal/tar/corretto-build/buildRoot/make/Init.gmk:192: images] Error 2
corretto11>
corretto11> > Task :installers:linux:universal:tar:executeBuild FAILED
corretto11>
corretto11> [Incubating] Problems report is available at: file:///build/source/corretto-build/reports/problems/problems-report.html
corretto11>
corretto11> FAILURE: Build failed with an exception.
corretto11>
corretto11> * What went wrong:
corretto11> Execution failed for task ':installers:linux:universal:tar:executeBuild'.
corretto11> > Process 'command 'make'' finished with non-zero exit value 2
corretto11>
corretto11> * Try:
corretto11> > Run with --stacktrace option to get the stack trace.
corretto11> > Run with --info or --debug option to get more log output.
corretto11> > Run with --scan to get full insights.
corretto11> > Get more help at https://help.gradle.org.
corretto11>
corretto11> Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
corretto11>
corretto11> You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
corretto11>
corretto11> For more on this, please refer to https://docs.gradle.org/8.14.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
corretto11>
corretto11> BUILD FAILED in 52s
corretto11> 10 actionable tasks: 10 executed
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.3, NixOS, 26.05 (Yarara), 26.05.20260102.fb7944c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+2`
 - nixpkgs: `/nix/store/as8ksm9hafq3x3cvml95i5s092903kvk-source`

### Notify maintainers

@rollf 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/compilers/corretto/11.nix
+++ b/pkgs/development/compilers/corretto/11.nix
@@ -20,12 +20,6 @@ let
       ;
     jdk = jdk11;
     gradle = gradle_8;
-    extraConfig = [
-      # jdk11 is built with --disable-warnings-as-errors (see openjdk/11.nix)
-      # because of several compile errors. We need to include this parameter for
-      # Corretto, too.
-      "--disable-warnings-as-errors"
-    ];
     version = "11.0.29.7.1";
     src = fetchFromGitHub {
       owner = "corretto";

--- a/pkgs/development/compilers/corretto/11.nix
+++ b/pkgs/development/compilers/corretto/11.nix
@@ -20,12 +20,12 @@ let
       ;
     jdk = jdk11;
     gradle = gradle_8;
-    version = "11.0.29.7.1";
+    version = "11.0.30.7.1";
     src = fetchFromGitHub {
       owner = "corretto";
       repo = "corretto-11";
       rev = version;
-      hash = "sha256-/VlV8tAo1deOZ5Trc4VlLNtpjWx352qUGZmfVbj7HuU=";
+      hash = "sha256-SUdJlTYE+RRAZa8DhFW0EYW1kHmuNDG+hk+/3MXtx1w=";
     };
   };
 in

--- a/pkgs/development/compilers/corretto/mk-corretto.nix
+++ b/pkgs/development/compilers/corretto/mk-corretto.nix
@@ -44,6 +44,7 @@ jdk.overrideAttrs (
     postPatch =
       let
         extra_config = builtins.concatStringsSep " " extraConfig;
+        jdk_configure_flags = "'" + builtins.concatStringsSep "', '" oldAttrs.configureFlags + "'";
       in
       (oldAttrs.postPatch or "")
       + ''
@@ -70,6 +71,12 @@ jdk.overrideAttrs (
           substituteInPlace $file --replace-warn "workingDir '/usr/bin'" "workingDir '.'"
         done
 
+        # Prepend corresponding OpenJDK flags to Corretto's own configure flags.
+        # This will provide us with the version-specific fixes (see
+        # openjdk/generic.nix) while giving Corretto's flags precedence.
+        # Note that the OpenJdK flags contain "--with-boot-jdk=..."!
+        substituteInPlace build.gradle --replace-fail "correttoCommonFlags = [" "correttoCommonFlags = [${jdk_configure_flags} ,"
+        # Finally, *append* nix-corretto-version specific flags.
         gradleFlagsArray+=(-Pcorretto.extra_config="${extra_config}")
       '';
 

--- a/pkgs/development/compilers/corretto/mk-corretto.nix
+++ b/pkgs/development/compilers/corretto/mk-corretto.nix
@@ -145,6 +145,7 @@ jdk.overrideAttrs (
       license = lib.licenses.gpl2Only;
       description = "Amazon's distribution of OpenJDK";
       maintainers = with lib.maintainers; [ rollf ];
+      platforms = lib.platforms.linux;
       teams = [ ];
     };
   }

